### PR TITLE
[IMP] website: add the email of when a user fill the website form

### PR DESCRIPTION
**Pull request by std-odoo**
Purpose
=======
When a public user fill a form on Website (e.g. /contactus), an email
will be sent. The email filled in the form will be used as the "email
from", but if no mail server match this email address it will be
encapsulated into "`notifications@mycompany.com`".

The email is still present in the "Reply-To" header, we want to add it
at the end of the email, so the receiver has this information easily.

Task-2833093

### CODE REVIEW

File: addons/website/controllers/form.py [hunk 1]
```diff
@@ -173,6 +173,10 @@ def extract_data(self, model, values):
                 except ValueError:
                     error_fields.append(field_name)
 
+                if dest_model._name == 'mail.mail' and field_name == 'email_from':
```
Comment (awa-odoo) [inline addons/website/controllers/form.py:176]: Any reason to put this outside of the try/except?

File: addons/website/controllers/form.py [hunk 2]
```diff
@@ -173,6 +173,10 @@ def extract_data(self, model, values):
...
+                    # Always add the email in the custom message
```
Comment (awa-odoo) [inline addons/website/controllers/form.py:177]: Could we elaborate a bit?
Something like:

> As the "email_from" is used to populate the email_from of the sent mail.mail,
> it could be filtered out at sending time (since #[reference-to-PR]).
> To make sure the email contains that (important) information, we also add it
> to the "custom message" that will be included in the body of the email sent.

(If I understood the issue properly).

File: addons/website/controllers/form.py [hunk 3]
```diff
@@ -173,6 +173,10 @@ def extract_data(self, model, values):
...
+                    custom_fields.append((_('email'), field_value))
```
Comment (awa-odoo) [inline addons/website/controllers/form.py:178]: Should we capitalize "Email" ?
Maybe not, not sure how it looks in the email body for other custom fields.

**Review by awa-odoo — COMMENTED** [review]
Hello @std-odoo 

The fix is looking good, I added a few suggestions.

May I ask why it is done in master and not in a stable version?
It does not seem like it would break the stable policy, you would just need to export the associated translation.

Also, I would rephrase the commit message a bit and change:

> The email is still present in the "Reply-To" header...

To:

> Even though the email is still present in the "Reply-To" header...

Feels like the sentence would make more sense that way.

In addition, we could also add a link to the PR that introduces this "mail server matching" change for reference in the commit message.

Thanks & cheers!

**Review by std-odoo — COMMENTED** [review]
@awa-odoo Hi :) 

I've done the changes :) 

I considered this as an improvement, but indeed we can consider it as a fix, so I close this PR and I did a new one here: https://github.com/odoo/odoo/pull/94653
